state/redis/redis.go

```diff
@@ -68,28 +69,28 @@ func (r *StateStore) Init(metadata state.Metadata) error {
 		return err
 	}
 
-	ctx := context.Background()
-	opts := redisconn.Opts{
-		DB:       0,
-		Password: redisCreds.Password,
-	}
-	conn, err := redisconn.Connect(ctx, redisCreds.Host, opts)
+	var tlsConfig *tls.Config
+	err = json.Unmarshal(b, tlsConfig)
 	if err != nil {
 		return err
 	}
 
-	r.client = &redis.SyncCtx{
-		S: conn,
+	opts := redis.Options{
+		DB:        0,
```

Comment:
it would need host, specify it as:
` Addr: redisCreds.Host`

Reply:
+1. Also we need to consider taking MaxRetries and MaxBackoff configuration.

Reply:
Oh right, I have added the host now. 



> +1. Also we need to consider taking MaxRetries and MaxBackoff configuration.

I have added those both now but how do we decide what all options should we consider?

Reply:
Thats fine.